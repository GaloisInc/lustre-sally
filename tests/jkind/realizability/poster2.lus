-- Unrealizable for ints
-- Unknown for reals (but should be realizable)

node main(in1, in2, out : int) returns (ok : bool);
let
  assert in1 < in2;
  ok = (in1 < out and out < in2);
  --%PROPERTY ok;
  --%REALIZABLE in1, in2;
tel;
