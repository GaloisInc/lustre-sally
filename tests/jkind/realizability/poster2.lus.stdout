Sally Model
===========

(define-state-type
   S
   (
     (Lt Bool)
     (in1 Int)
     (in2 Int)
     (ok Bool)
     (out Int)
     (|gal-initializing| Bool))
   (
     (in1 Int)
     (in2 Int)
     (out Int)))
(define-transition-system
   TS
   S
   (= |gal-initializing| true)
   (and
      (= next.|gal-initializing| false)
      (= next.in1 input.in1)
      (= next.in2 input.in2)
      (= next.out input.out)
      next.Lt
      (=
         next.ok
         (and
            (< next.in1 next.out)
            (< next.out next.in2)))
      (=
         next.Lt
         (< next.in1 next.in2))))

(query
   TS
   (or |gal-initializing| ok))

[Lustre] Validating properties:
[Lustre] Property ok... [Invalid] 
Trace
=====

ok:
 Step | in1 | in2 | out | |-> | ok    
------+-----+-----+-----+-----+-------
 1    | 0   | 1   | 0   |     | False 

[Lustre] Model status: [Invalid] 
